docs/source/blogs/tech_blog/blog12_Inference_Time_Compute_Implementation_in_TensorRT-LLM.md

```diff
@@ -0,0 +1,425 @@
+            self.tokenizer.encode(self.probe_suffix, add_special_tokens=False))
+
+        current_prompt = initial_prompt
+```
+
+To prevent extra latency, the `proposer_task` should not be blocked by the `probe_task`. Scaffolding's task-level concurrency handles this perfectly. We can yield `proposer_task` and `probe_task` in a single list. Multiple tasks yielded together in the same list will be batched and executed in parallel.
+
+```python
+    yield[proposer_task, probe_task]
+```
+
+In the following `for` loop, each iteration performs these steps:
+
+1. **Submit** both a proposer task and a probe task by yielding them. We don't need to worry about execution details, as they are handled by `ScaffoldingLlm`, which binds the `Controller` and `Workers` together behind the scenes.
+
+2. **Evaluate** the probe response after the tasks return, checking for consistency over several rounds (using `certainty_threshold`).
+
+3. **Finalize** the answer and return if it is consistent. Otherwise, append the new tokens from the proposer task and proceed to the next iteration.
+
+```python
+        # Iterate over generation rounds until the maximum tokens limit is reached.
+        # Make sure length of prefilling is always smaller than the max_tokens in TRTLLMWorker.init_with_new_llm
+        # Otherwise it will through an assertion fail, stated in issue #3576
+        for _ in range(initial_prompt_token_num + probe_suffix_token_num,
+                    self.max_tokens, self.chunk_size):
+            proposer_task.input_str = current_prompt
+            probe_task.input_str = current_prompt + self.probe_suffix
+
+            # For the probe task, append the suffix to force a chain-of-thought leading to an answer.
+            yield [proposer_task, probe_task]
+
+            # Retrieve the output from the probe task.
+            probe_text = probe_task.output_str
+
+            # Extract the potential answer from the probe response.
+            answer = self.obtain_answer(probe_text)
+            probe_answers.append(answer)
+            probe_responses.append(probe_text)
+
+            if self.should_early_stop(probe_answers, probe_responses):
+                tasks[0].result = probe_task.result
+                # If the current prompt indicates the chain-of-thought phase has ended, use one type of suffix.
+                if "</think>" in current_prompt:
+                    tasks[0].output_str = (current_prompt + self.answer_suffix +
+                                        probe_answers[-1] + "}\n\\]")
+                    return
+                else:
+                    # Otherwise, use the suffix with marker to transition clearly.
+                    tasks[0].output_str = (current_prompt +
+                                        self.answer_suffix_with_marker +
+                                        probe_answers[-1] + "}\n\\]")
+                    return
+
+            # If not confident, do another round of generation
```

Comment:
suggestion: If the answer is not deemed confident, perform another round of generation.